src/lib/node/message_handler.ts

```diff
@@ -73,9 +79,14 @@ export class NodeMessageHandler {
           actualFirmware.data,
           actualFirmware.firmwareTarget
         );
+        this.firmwareUpdateQueued[nodeId] = true;
+        node.once("firmware update progress", (_, __, ___) => {
+          this.firmwareUpdateQueued[nodeId] = false;
+        });
```

Comment:
Instead of using `once`, maybe you can check if the listener has been attached once, and if not, attach it:

```ts
if (!(nodeId in this.firmwareUpdateQueued)) {
  node.addEventListener(...)
}
this.firmwareUpdateQueued[nodeId] = true;
```